junit/mockwebserver/src/main/java/io/fabric8/mockwebserver/dsl/Function.java

```diff
@@ -0,0 +1,21 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.fabric8.mockwebserver.dsl;
+
+public interface Function<I, O> {
+  O apply(I input);
```

Comment:
Umm, can use  `java.util.function` here?

Reply:
I just copied the code from the original repository.

The only things that I've changed are those related to the certificates used by the server in TLS mode (as discussed internally).

There are plenty of things to refactor. Also, a lot of duplicate code from utils that are already present in the kubernetes-client-api module. We might need to see if we create a shared module for those so that they can be reused.

In any case, I'd rather do any refactoring in subsequent PRs so we have a snapshot starting point that is the same as what we have in the original repository.